python/python/bystro/rare_variant/parent_of_origin.py

```diff
@@ -0,0 +1,178 @@
+"""
+This module implements a parent of origin effect caller. The parent of 
+origin effect refers to the difference in phenotypes 
+depending on whether the allele is inherited paternally or maternally. The 
+caller employs a method that draws inspiration from the community 
+detection problem studied in computer science.
+
+The distribution of phenotypes in the heterozygotes is modeled as a 
+Gaussian mixture model. It assumes identical covariance matrices matching 
+the homozygous population but with distinct means for the maternal and 
+paternal populations. The model identifies this difference as the 
+principal eigenvector of the heterozygote population, after undergoing a 
+transformation that "whitens" the homozygous population. This process 
+corresponds to finding the dimension along which the covariance is 
+"stretched" compared to the heterozygote population.
+
+A justification for this method can be found in Vershynin's "High-
+Dimensional Probability" Chapter 4.7 or Wainwright's "High-Dimensional 
+Statistics" Chapter 8.1.
+
+Methods
+-------
+get_low_rank(A)
+    This extracts the rank-1 matrix approximation to A using the singular
+    value decomposition.
+
+Objects
+-------
+BasePOE
+    The base class of the parent of origin effect caller. Currently just 
+    implements methods to test inputs for proper dimensionality and a
+    method to classify heterozygotes based on their phenotypes
+
+POESingleSNP(BasePOE)
+    A caller for the POE of a single SNP. No sparsity assumptions are 
+    implemented in this model.
+"""
+import numpy as np
+import numpy.linalg as la
+
+
+def get_low_rank(A):
```

Comment:
Could you include a test for this function? It's not clear from the code/tests alone the relationship between get_low_rank and POESingleSNP and how they're meant to interact when using this method.

Reply:
Ah residual no longer needed. Will remove.